mbed_tools_ci_scripts/tag_and_release.py

```diff
@@ -68,15 +70,20 @@ def _update_documentation() -> None:
     Pdoc nests its docs output in a folder with the module's name.
     This process removes this unwanted folder.
     """
-    docs_dir, docs_contents_dir = _get_documentation_paths()
+    docs_dir, module_to_document = _get_documentation_config()
+    clear_previous_docs(docs_dir)
+    os.makedirs(str(docs_dir), exist_ok=True)
     with TemporaryDirectory() as temp_dir:
-        shutil.move(str(docs_contents_dir), str(temp_dir))
-        shutil.rmtree(docs_dir)
-        shutil.move(str(temp_dir), str(docs_dir))
+        generate_docs(temp_dir, module_to_document)
+        docs_contents_dir = temp_dir.joinpath(module_to_document)
+        if docs_contents_dir.exists() and docs_contents_dir.is_dir():
+            for element in docs_contents_dir.iterdir():
```

Comment:
After talking IRL - this might be better in generate docs so we never have to worry about the module-name wrapper directory.